### ISSUE TITLE
Incorrect error generated when comparing enum values.

### ISSUE THREAD

**Issue by dbaeumer** [Microsoft/TypeScript#10989]
**TypeScript Version:**  2.0.2
**Code**

```ts
enum Direction {
	LEFT,
	RIGHT
}

function bug(direction?: Direction): number {
	if (!direction) {
		return 10;
	} else {
		return (direction === Direction.LEFT) ? 10 : 0;
	}
}
```

You get a compile error saying the Direction can't be used in ===. This is IMO incorrect.

**Comment by mhegazy** [comment]
`Direction.LEFT` is `0`, it will be filtered out in `!direction` check. so the only value `direction` could have in the `else` clause is `Direction.RIGHT`; and `Direction.RIGHT` will never be `Direction.LEFT`.

So depending on what is the intent here, either make `LEFT` non-false:
```ts
enum Direction {
    LEFT = 1,
    RIGHT
}
```

or just trust that this will produce the same result:

```ts
    if (!direction) {
        return 10;
    } else {
        return  0;
    }
```

**Comment by dbaeumer** [comment]
You are absolutely right and that error actually catches a problem in our code. JS falsy ....

**Comment by dbaeumer** [comment]
Closing.

**Comment by basarat** [comment]
> JS falsy

I've had this happen to be even before it was an error. Hence : https://basarat.gitbooks.io/typescript/content/docs/enums.html#changing-the-number-associated-with-an-enum I added `I quite commonly initialize the first enum with = 1 as it allows me to do a safe truthy check on an enum value.` Can help fix that error quickly ;) :rose:

**Comment by forthedamn** [comment]
I encountered the same situation, and solved in same way.

> initialize the first enum with = 1

That will be great!
